src/main/java/org/primefaces/renderkit/HeadRenderer.java

```diff
@@ -123,6 +127,11 @@ public void encodeBegin(FacesContext context, UIComponent component) throws IOEx
             encodeValidationResources(context, applicationContext.getConfig().isBeanValidationEnabled());
         }
 
+        if (applicationContext.getConfig().isClientSideLocalizationEnabled()) {
+            Locale locale = LocaleUtils.getCurrentLocale(context);
+            encodeJS(context, "locales/locale-" + locale.getLanguage() + ".js");
```

Comment:
Should the user make sure that all their application locale files are present? I would say so. Something to point out in the documentation.

Reply:
I actually just updated to throw an exception. If you are using CLIENT_SIDE_LOCALISATION and your file does not exist we should throw a FacesException like the encodeCSS function does if the CSS is missing.